### PR TITLE
Add border radius to buttons

### PR DESCRIPTION
**Pull request by juanfra**
🎫 https://central.tri.be/issues/129297

Adding border radius to buttons to match new designs.

### CODE REVIEW

File: mixins/_buttons.pcss [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 @define-mixin button-global {
 	border: 0;
+	border-radius: 4px;
```
Comment (paulmskim) [inline mixins/_buttons.pcss]: The `button-global` is a button reset, we should apply a `border-radius` to each of the button styles that use it. If later we define another button style without a border radius, this won't work.

Comment (juanfra) [reply mixins/_buttons.pcss]: Hey Paul, thanks for the code review. The whole design system changed and they decided to use curved border buttons. For this design system, the default button is one with curved edges. I guess in the case they later add a squared button, we can add a variation of the class with a modifier for that.

Comment (paulmskim) [reply mixins/_buttons.pcss]: Right, I understand. What I'm saying is that we should be adding the `border-radius` to the `button-solid` and `button-border` mixins.

Comment (juanfra) [reply mixins/_buttons.pcss]: I got you wrong then. I thought you wanted to define it as something different and then in common define it for each button we define there.

**Pull request merged**